### ISSUE TITLE
Ground Sloth

### ISSUE THREAD

**Issue by KarBoy2314PL** [StardewModders/mod-ideas#243]
from #132 

If you complete a community center bundle that rewards you with a sloth egg (Shhhh its a mystery egg and no one knows why its an mammal in an egg its just Junimo magic) The next morning Robin mails you a letter saying she heard about the egg and that she can build a special building for it to incubate in as it is rather large, in 5 days after the building is built and in another 10-15 days the egg hatches and a Ground sloth can be ridden but it does something special... It has its own energy bar that can be used to clear rocks and debris rather quickly,

TL;DR - A new egg, when incubated you get a ridable animal that behaves like a tractor but has limited energy

**Comment by Pathoschild** [comment]
This is a pretty complex mod request. It involves...
* adding a custom item (sloth egg);
* adding a custom farm animal (sloth);
* adding a custom building (sloth house?);
* adding a custom location (sloth house interior);
* overriding the egg incubator logic to work with sloth eggs;
* replicating features from the Tractor Mod;
* adding a separate energy bar with associated tracking logic;
* handling save persistence logic (e.g. need to remove everything custom before save to avoid breaking the save if the mod is removed).

All of those things individually are possible though, it's just a large task.
